docs/guide/resources/synthetic_dataset_generation.md

```diff
@@ -0,0 +1,65 @@
+Few shot learning is one of the most powerful tools that autolabel offers to improve the accuracy of LLM generated labels. However, curating a seed dataset to use for few shot learning can be a time consuming and tedious process. To make this process easier, autolabel's LabelingAgent provides a method to generate synthetic datasets. These datasets can be used as seed datasets for few shot learning or for any other purpose. This guide will walk you through the process of generating a synthetic dataset using autolabel.
+
+Currently, autolabel supports synthetic dataset generation for classification and entity matching tasks. We plan to add support for other task types in the future.
+
+### **Walkthrough: Creating a Synthetic Dataset for Banking**
+
+<ol>
+<li>The first step is to import the LabelingAgent from autolabel. This is the main class that we will use to generate the synthetic dataset.
+
+```python
+from autolabel import LabelingAgent
+```
+
+</li>
+<li>The next step is to create the task config. Make sure to add the <code>dataset_generation</code> section to the config. This section contains the parameters for the dataset generation process. The <code>guidelines</code> parameter is a string containing the guidelines for the dataset generation task. The <code>num_rows</code> parameter is an integer indicating the number of rows <em><strong>per label</strong></em> to generate in the dataset.
+
+```python
+config = {
+  "task_name": "BankingComplaintsClassification",
+  "task_type": "classification",
+  "dataset": {
+    "label_column": "label",
+    "delimiter": ","
+  },
+  "model": {
+    "provider": "openai",
+    "name": "gpt-3.5-turbo"
+  },
+  "prompt": {
+    "task_guidelines": "You are an expert at understanding bank customers support complaints and queries.\nYour job is to correctly classify the provided input example into one of the following categories.\nCategories:\n{labels}",
+    "output_guidelines": "You will answer with just the the correct output label and nothing else.",
+    "labels": [
+      "activate_my_card",
```

Comment:
how are we adding label descriptions

Reply:
We aren't in this case, which would just provide the labels to the LLM without descriptions. I did forget to add it to the docs though so thanks for reminding me!